### ISSUE TITLE
dns: error: upstream nameservers not set

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#2568]
My macOS, running 1.13.19, on `tailscale.com`, is logging:

```
7264	default	09:35:15.588885-0700	IPNExtension	dns: error: upstream nameservers not set
7264	default	09:35:15.589015-0700	IPNExtension	dns: error: upstream nameservers not set
```

/cc @crawshaw 

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_3ikpd)

**Comment by darshinimashar** [comment]
iOS : reporting error: upstream nameservers not set as well and having DNS issue. Also macOS has the same on 1.14 version

**Comment by DentonGentry** [comment]
The user in the linked support conversation reported this problem fixed in 1.14.0

**Comment by zeha** [comment]
I see this on Linux with tailscale 1.24.2, and `--accept-dns=false`. Don't quite understand why tailscale wants to do any DNS things in this case.

**Comment by DentonGentry** [comment]
Is DNS working on that system, are you able to get names resolved?

The log message means tailscaled thinks it received a DNS packet which it could not forward anywhere because there were no resolvers (because `--accept-dns=false`). https://github.com/tailscale/tailscale/blob/9f3ad407078e5f0e1a2658a4c8abbe4acfb63b2b/net/dns/resolver/forwarder.go#L645

Why it thinks it received a DNS packet isn't clear; I just set `tailscale up --accept-dns=false` on a Debian system as a quick check and don't see any such message.

A couple things could be checked:
- does /etc/resolv.conf contain 'nameserver 100.100.100.100`? It shouldn't but maybe something weird happened
- if you are willing to run `apt install tcpdump` and `tcpdump -i tailscale0 -v port 53` it might show if DNS packets really are arriving

**Comment by zeha** [comment]
DNS works, yes. Obviously the tailnet names do not work, but thats expected.

> does /etc/resolv.conf contain nameserver 100.100.100.100? It shouldn't but maybe something weird happened

It only has:
```
nameserver 127.0.0.1
search home.zeha.at
```

127.0.0.1:53 is a PowerDNS Recursor.

> if you are willing to run apt install tcpdump and tcpdump -i tailscale0 -v port 53 it might show if DNS packets really are arriving

Ok, wow... indeed there were DNS requests arriving here. Looks like another long-running process cached an old copy of `/etc/resolv.conf` and always talked to `100.100.100.100:53` - all of those queries failed, and were retried every 5 sec.
Maybe if the tailscaled error message would have said additional things (say a client IP or DNS query labels) I would have thought of that earlier?

**Comment by DentonGentry** [comment]
Original issue resolved in 1.14.
